sklearn/decomposition/nmf.py

```diff
@@ -277,30 +632,208 @@ class ProjectedGradientNMF(BaseEstimator, TransformerMixin):
             'nndsvdar': NNDSVD with zeros filled with small random values
                 (generally faster, less accurate alternative to NNDSVDa
                 for when sparsity is not desired)
-            'random': non-negative random matrices
+            'custom': use custom matrices W and H
+
+    update_H : boolean, default: True
+        Set to True, both W and H will be estimated from initial guesses.
+        Set to False, only W will be estimated.
+
+    solver : 'proj-grad' | 'coordinate'
```

Comment:
`DictionaryLearning` uses `fit_algorithm : {‘lars’, ‘cd’}` so I'm +1 for `projgrad|cd` or `pg|cd` here. 
Not sure about dashes for separators here, but if we want full names I'd do `projected_gradient|coordinate_descent`. Anything we want to be future-proof about here?